project-codeflare/codeflare-sdk
# update SDK args

# Issue link


[RHOAIENG-7200](https://issues.redhat.com/browse/RHOAIENG-7200)
# What changes have been made


What

Update the Cluster configuration in the SDK to make arguments clearer.

Currently the default config is as such:
```
cluster = Cluster(ClusterConfiguration(
    name='jobtest',
    namespace='default', # Update to your namespace
    num_workers=2,
    min_cpus=1,
    max_cpus=1,
    min_memory=4,
    max_memory=4,
    num_gpus=1,
    head_gpus=1,
    image="quay.io/project-codeflare/ray:latest-py39-cu118",
    write_to_file=False, # When enabled Ray Cluster yaml files are written to /HOME/.codeflare/resources 
    local_queue="local-queue-test" # Specify the local queue manually
)) 
```

How

Alter the following args:

* head_gpus -> num_head_gpus
* num_gpus ->  num_worker_gpus
* min_cpus -> worker_cpu_requests
* max_cpus -> worker_cpu_limits
* min_memory -> worker_memory_requests
* max_memory -> worker_memory_limits

# Verification steps
## Setup
### Notebook server ODH/RHOAI/Local
* Clone this repository with `git clone https://github.com/project-codeflare/codeflare-sdk.git`
* Checkout this PR's branch
* Run `poetry build` - install if needed (`pip install poetry`)
* Run `pip install --force-reinstall dist/codeflare_sdk-0.0.0.dev0-py3-none-any.whl` 
* Restart your notebook kernel
## Manual Testing
* Open any of the guided demo notebooks.
* Follow step by step using the old arguments, warnings should be thrown when ClusterConfiguration is run.
* Rerun the notebook using the new, suggested arguments, no errors should be given.
## Checks
- Testing Strategy
